src/main.rs

```diff
@@ -1,15 +1,183 @@
 mod comms;
 mod stableset;
 
-use crate::comms::{Comm, NetworkNode};
+use bytes::Bytes;
+use net::connection::Connection;
+use net::endpoint::Endpoint;
+use tokio::sync::RwLock;
+use tokio::sync::mpsc::Receiver;
+
+use crate::comms::{Comm, NetworkNode, NetworkMsg, MsgId};
 use crate::stableset::{run_stable_set, StableSetMsg};
 
-use std::collections::BTreeSet;
-use std::{env, fs, net::SocketAddr, time::Duration};
-use tokio::runtime::Runtime;
+use std::collections::{BTreeMap, BTreeSet};
+use std::sync::Arc;
+use std::{env, fs, net::SocketAddr};
 
 const PEERS_CONFIG_FILE: &str = "peers.json";
 
+mod net;
+
+#[derive(Clone)]
+struct MyComm {
```

Comment:
I think this network specific code should be in another module, not in main. Don't you think?